### ISSUE TITLE
At entry to EODAD routine R14 is incorrect

### ISSUE THREAD

**Issue by abekornelis** [z390development/z390#525]
At entry to EODAD routine R14 should point to the resume address so that the EODAD can be exited with a BR R14 in order to resume the main program logic immediately after the SVC that caused a READ/GET attempt.
Currently the R14 content remains unchanged from the main program logic and BR R14 from the EODAD will take the program to whatever address happens to be in there :-(

**Comment by jyganci** [comment]
A GET macro issues SVC 151. The code in sz390 for SVC 151 is get_svc(). A quick look at the code does, indeed, verify that register 14 is not set when end of input is detected. See sz390 method dcb_eodad_exit(). Been this way since the beginning.

Look at demo/TESTDCB1.MLC to see how it handles EODAD. It appears that, at the present time, this is the only way for a user to handle EODAD.

First guess at fix is for sz390 dcb_eodad_exit() code, when modifying the PSW to point to the EODAD address, should also set register 14 to the address in the "SVC OLD PSW", which should be the instruction following the GET. There is no "SVC OLD PSW" field. The current PSW, maintained in pz390, is what would've been stored in "SVC OLD PSW". The address of the instruction following the SVC 151 is in pz390 field psw_loc. This could be used to set register 14 in dcb_eodad_exit() just before the code there issues pz390.set_psw_loc(cur_dcb_eodad).

Quick one line fix for sz390.dcb_eodad_exit() shown in context. Lines at end of the method.

```
	...
	if (cur_dcb_eodad == 0){
		abort_error(31,"read at end of file and no EODAD for " + tiot_ddnam[cur_tiot_index]);
	} else {
		pz390.reg.putInt((14<<3)+4,pz390.psw_loc);  // #525
		pz390.set_psw_loc(cur_dcb_eodad);
	}
```
This has been tested reading a file with an EODAD routine that sets an EOF flag and exits via BR 14. Seems to work.

TODO: investigate further.

Similar comments apply to READ,CHECK. A CHECK issues SVC 155. The sz390 code for SVC 155 is svc_check(), which also invokes dcb_eodad_exit(). The above one line fix applies to CHECK as well.

**Comment by jyganci** [comment]
Fixed by PR #528.
